config/database.yml

```diff
@@ -1,27 +1,52 @@
+<%
+  # first env with a value
+  def first_of(*envs)
+    envs.map { |env| ENV[env] }.compact.first
+  end
+
+  def puts_red(msg)
+    puts "\e[31m#{msg}\e[0m"
+  end
+
+  def deprecated(env, message)
+    puts_red "DEPRECATED: #{env} is deprecated, #{message}" if ENV[env]
+  end
+
+  deprecated("HOSTEDGPT_DATABASE_PORT", "use HOSTEDGPT_DB_PORT instead")
+  deprecated("HOSTEDGPT_DEV_DB", "use HOSTEDGPT_DB_NAME instead")
+  deprecated("HOSTEDGPT_TEST_DB", "provide HOSTEDGPT_DB_NAME and test database name is generated")
+  deprecated("HOSTEDGPT_DATABASE_PASSWORD", "use HOSTEDGPT_DB_PASSWORD instead")
+%>
+
 default: &default
   adapter: postgresql
+  # url format: postgres://username:password@localhost:5432/database_name
+  url: <%= ENV["DATABASE_URL"] %>
+  # It's best to set DATABASE_URL to configure database details. That overrides the following settings
+  username: <%= ENV["HOSTEDGPT_DB_USERNAME"] || "app" %>
+  password: <%= first_of("HOSTEDGPT_DATABASE_PASSWORD", "HOSTEDGPT_DB_PASSWORD") %>
```

Comment:
Is first_of any different than `ENV[..] || ENV[..]`? It's fine to leave it in, I just couldn't tell if there was an edge case you were running into with that approach which first_of was solving for

Reply:
yeah it should be the same. I can remove it to avoid the extra code. I thought it would be more helpful than it ended up being